napalm/eos/eos.py

```diff
@@ -672,140 +673,84 @@ def get_interfaces_counters(self):
             )
         return interface_counters
 
-    def get_bgp_neighbors(self):
-        def get_re_group(res, key, default=None):
-            """Small helper to retrieve data from re match groups"""
-            try:
-                return res.group(key)
-            except KeyError:
-                return default
-
-        NEIGHBOR_FILTER = "vrf all | include IPv[46] (Unicast|6PE):.*[0-9]+ | grep -v ' IPv[46] Unicast:/.' | remote AS |^Local AS|Desc|BGP state |remote router ID"  # noqa
-        output_summary_cmds = self._run_commands(
-            ["show ipv6 bgp summary vrf all", "show ip bgp summary vrf all"],
-            encoding="json",
-        )
-        output_neighbor_cmds = self._run_commands(
+    def get_bgp_neighbors(self) -> Dict[str, models.BGPStateNeighborsPerVRFDict]:
+        cmd_outputs = self._run_commands(
             [
-                "show ip bgp neighbors " + NEIGHBOR_FILTER,
-                "show ipv6 bgp peers " + NEIGHBOR_FILTER,
+                "show ip bgp summary vrf all",
+                "show ipv6 bgp summary vrf all",
+                "show ip bgp neighbors vrf all",
+                "show ipv6 bgp peers vrf all",
             ],
-            encoding="text",
+            encoding="json",
         )
 
-        bgp_counters = defaultdict(lambda: dict(peers={}))
-        for summary in output_summary_cmds:
-            """
-            Json output looks as follows
-            "vrfs": {
-                "default": {
-                    "routerId": 1,
-                    "asn": 1,
-                    "peers": {
-                        "1.1.1.1": {
-                            "msgSent": 1,
-                            "inMsgQueue": 0,
-                            "prefixReceived": 3926,
-                            "upDownTime": 1449501378.418644,
-                            "version": 4,
-                            "msgReceived": 59616,
-                            "prefixAccepted": 3926,
-                            "peerState": "Established",
-                            "outMsgQueue": 0,
-                            "underMaintenance": false,
-                            "asn": 1
-                        }
-                    }
-                }
-            }
-            """
-            for vrf, vrf_data in summary["vrfs"].items():
-                bgp_counters[vrf]["router_id"] = vrf_data["routerId"]
-                for peer, peer_data in vrf_data["peers"].items():
-                    if peer_data["peerState"] == "Idle":
+        bgp_counters = defaultdict(
+            lambda: models.BGPStateNeighborsPerVRFDict(
+                peers=models.BGPStateNeighborDict()  # type: ignore
+            )  # type: ignore
+        )  # type: ignore
+        # Iterate IPv4 and IPv6 neighbor details
+        for cmd in cmd_outputs[2:]:
```

Comment:
The idea is we want to run 4 commands.  Rather than doing 2 roundtrips, we do all 4 at once (ipv4 bgp sum, ipv6 bgp sum, ipv4 bgp neigh, ipv6 bgp neigh) and then process the pairs (both bgp sum, both bgp neigh)